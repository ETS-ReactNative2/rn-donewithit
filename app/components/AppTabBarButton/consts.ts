import COLORS from '../../config/colors'

const CONSTS = {
  TabBarButtonTop: -20,
  iconSize: 30,
  iconColor: COLORS.WHITE,
  iconName: 'plus-circle',
  firstCircleSize: 60,
  secondCircleSize: 50
}

export default CONSTS
