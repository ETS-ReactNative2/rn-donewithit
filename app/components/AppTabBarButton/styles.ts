import { StyleSheet, Platform } from 'react-native'
import COLORS from '../../config/colors'
import CONSTS from './consts'

const styles = StyleSheet.create({
  container: {
    top: CONSTS.TabBarButtonTop
  },
  firstCircle: {
    ...Platform.select({
      android: {
        elevation: 1
      },
      ios: {
        shadowOpacity: 1
      }
    }),
    backgroundColor: COLORS.WHITE,
    borderRadius: CONSTS.firstCircleSize / 2,
    width: CONSTS.firstCircleSize,
    height: CONSTS.firstCircleSize,
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: COLORS.PRIMARY
  },
  secondCircle: {
    backgroundColor: COLORS.PRIMARY,
    borderRadius: CONSTS.secondCircleSize / 2,
    width: CONSTS.secondCircleSize,
    height: CONSTS.secondCircleSize,
    justifyContent: 'center',
    alignItems: 'center'
  }
})

export default styles
