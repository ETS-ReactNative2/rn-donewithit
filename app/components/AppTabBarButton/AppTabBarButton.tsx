import React from 'react'
import { View, TouchableOpacity } from 'react-native'
import AppIcon from '../AppIcon'
import styles from './styles'
import CONSTS from './consts'

type AppTabBarButtonProps = {
  iconName?: string
  iconSize?: number
  iconColor?: string
  onPress?: () => void
}

function AppTabBarButton({
  iconName = CONSTS.iconName,
  iconSize = CONSTS.iconSize,
  iconColor = CONSTS.iconColor,
  onPress
}: AppTabBarButtonProps) {
  return (
    <TouchableOpacity
      activeOpacity={0.85}
      style={styles.container}
      onPress={onPress}
    >
      <View style={styles.firstCircle}>
        <View style={styles.secondCircle}>
          <AppIcon name={iconName} size={iconSize} color={iconColor} />
        </View>
      </View>
    </TouchableOpacity>
  )
}

export default AppTabBarButton
