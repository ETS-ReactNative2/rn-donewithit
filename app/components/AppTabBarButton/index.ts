import AppTabBarButton from './AppTabBarButton'
export default AppTabBarButton
