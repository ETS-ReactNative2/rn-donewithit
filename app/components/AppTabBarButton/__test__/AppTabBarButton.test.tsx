import React from 'react'
import renderer from 'react-test-renderer'
import AppTabBarButton from '../index'

jest.mock(
  '@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js',
  () => {
    return () => ''
  }
)

const props = {
  iconName: 'email',
  iconSize: 25,
  iconColor: 'red'
}

describe('AppTabBarButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppTabBarButton {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
