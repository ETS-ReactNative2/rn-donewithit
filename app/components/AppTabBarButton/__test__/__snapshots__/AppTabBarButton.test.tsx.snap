// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTabBarButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "top": -20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 30,
        "height": 60,
        "justifyContent": "center",
        "shadowColor": "tomato",
        "shadowOpacity": 1,
        "width": 60,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "tomato",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
